packages/trader/src/AppV2/Components/TradeParameters/GrowthRate/growth-rate-picker.tsx

```diff
@@ -0,0 +1,101 @@
+}: TGrowthRatePickerProps) => {
+    const initial_growth_rate = React.useRef<number>();
+    const selected_growth_rate = React.useRef<number>(growth_rate);
+    const data = accumulator_range_list.map(rate => ({ value: `${getGrowthRatePercentage(rate)}%` }));
+    const details_content = [
+        {
+            label: <Localize i18n_default_text='Barrier' />,
+            value: `±${tick_size_barrier_percentage}`,
+        },
+        {
+            label: <Localize i18n_default_text='Max duration' />,
+            value: `${maximum_ticks || 0} ${maximum_ticks === 1 ? localize('tick') : localize('ticks')}`,
+        },
+    ];
+
+    React.useEffect(() => {
+        if (!initial_growth_rate.current && growth_rate) {
+            initial_growth_rate.current = growth_rate;
+        }
+        return () => {
+            if (initial_growth_rate.current && initial_growth_rate.current !== selected_growth_rate.current) {
+                setGrowthRate(initial_growth_rate.current);
+            }
+            debouncedSetGrowthRate.cancel();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const handleSave = () => {
+        initial_growth_rate.current = selected_growth_rate.current;
+    };
+
+    const handlePickerValuesChange = (value: string | number) => {
+        const new_value = Number((value as string).slice(0, -1)) / 100;
+        if (new_value === selected_growth_rate.current) return;
+        debouncedSetGrowthRate(setGrowthRate, new_value);
+        selected_growth_rate.current = new_value;
+    };
+
+    return (
+        <React.Fragment>
+            <ActionSheet.Content className='growth-rate__picker'>
+                <div className='growth-rate__wheel-picker'>
+                    {accumulator_range_list.length ? (
+                        <WheelPicker
+                            data={data}
+                            selectedValue={`${getGrowthRatePercentage(selected_growth_rate.current)}%`}
+                            setSelectedValue={handlePickerValuesChange}
+                        />
+                    ) : (
+                        <Skeleton />
+                    )}
+                </div>
+                <div className='growth-rate__details'>
+                    {details_content.map(({ label, value }) => (
+                        <span key={value} className='growth-rate__details-item'>
+                            <Text size='sm'>{label}</Text>
+                            {should_show_details ? <Text size='sm'>{value}</Text> : <Skeleton height={22} width={75} />}
+                        </span>
+                    ))}
+                </div>
+            </ActionSheet.Content>
+            <ActionSheet.Footer
+                alignment='vertical'
+                primaryAction={{
+                    content: <Localize i18n_default_text='Save' />,
+                    onAction: handleSave,
+                }}
+            />
+        </React.Fragment>
+    );
```

Comment:
note: moving forward, we will have muliple components with the same logic for wheel picker as we got here, so we will create a reusable TradeParamWheelPicker later on.

Reply:
![image](https://github.com/user-attachments/assets/f8af2fd8-037d-48a1-ba75-c5188dcf06ac)